src/spikeinterface/core/core_tools.py

```diff
@@ -934,3 +941,72 @@ def convert_bytes_to_str(byte_value: int) -> str:
         byte_value /= 1024
         i += 1
     return f"{byte_value:.2f} {suffixes[i]}"
+
+
+def spike_vector_to_dict(spike_vector: np.ndarray) -> list[dict]:
+    """
+    Computes all spike trains for all units/segments from a spike vector.
+
+    Parameters
+    ----------
+    spike_vector: np.ndarray
+        The spike vector to convert.
+
+    Returns
+    -------
+    spike_trains: list[dict]:
+        A list containing, for each segment, the spike trains of all units.
+    """
+    assert HAVE_NUMBA, "spike_vector_to_dict() requires `numba`!"
+
+    spike_trains = _vector_to_dict(
+        spike_vector["sample_index"].astype(np.int64),
```

Comment:
You should do `copy=False` in as type to avoid the copying when is not necessary. Otherwise this is doing copies all the time:

https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html

Reply:
Copying is necessary in some cases (for example, the spike vector of the `NumpySorting` is a `memmap`, which is incompatible in numba if you enforce the type).

I tried not enforcing the type and running with `copy=False`, and this is ~70% slower on my machine with my sorting object.

Reply:
The suggestion above is to avoid a redundant copy if the types are the same. Currently, if your function is already `np.int64` you will generate a copy anyway.

 If copying is necessary use an explicit function for copying. Do not rely on the implicit copying of `astype`

Reply:
The memmap is of type `np.int64`, but it needs to be a `np.ndarray`!
That's a two for one special with the `astype` ^^

Reply:
> The memmap is of type np.int64, but it needs to be a np.ndarray!

I think that you are equivocating numeric type with data-type. np.int64 is a numeric data type, np.ndarray is a datatype. Numpy memmaps are np.ndarray and can also have different numeric data types.

Reply:
I am actually curios about what error you are getting when you pass a memmap to numba.
I am also curios how you spike_vector ended up being a memmap.

Reply:
> I am actually curios about what error you are getting when you pass a memmap to numba.

`TypeError: No matching definition for argument type(s) readonly array(int64, 1d, A), readonly array(int64, 1d, A), readonly array(int64, 1d, A)`

> I am also curios how you spike_vector ended up being a memmap.

I think this is the default for the `NumpySorting`?

Reply:
Thanks for sharing the error. I was not aware of this:
You should do:
```python
array = np.array(array) if isintance(array, np.memmap) else array
```
This will avoid copy when it is not necessary.

It is not the default of NumpySorting. There are many constructors to the object.

Reply:
> Thanks for sharing the error. I was not aware of this

I also wasn't until today ^^

> You should do:
> `array = np.array(array) if isintance(array, np.memmap) else array`

I can do that. But if the dtype isn't `np.int64`, are you sure that `copy=False` is the way to go?
It will change the vector, which may not be what the user wants?

Reply:
Are you modifying the array inside the funciton?

Reply:
If you do `.astype(np.int64, copy=False)` then it changes the array no?
Or have I misunderstood something?

Reply:
Ah I just read the documentation, and you are right!
`copy=False` only disables copying if everything is correct.

I'll fix that :)

Reply:
Glad you read the documentation.